### DISCUSSION TITLE
Why is RecoilSync exposed as a Provider-like component while it's actually a singleton-like hook?

### DISCUSSION THREAD

**Discussion by FezVrasta** [facebookexperimental/Recoil#2038] [General]
I'm trying to understand why this design decision was made, as it often trips me up making me think anything child of a specific RecoilSync component will have some kind of context-like knowledge of its parent, while it doesn't seem to be the case.

**Comment by drarmstr** [comment]
It's due to the required semantics of the component.  `<RecoilSync>` should be an ancestor component that encompasses components which use the atoms that are synced with that store.  If it was just a hook then the hook could potentially be called after the nested component using the synced atom was called.

**Reply by FezVrasta** [reply]
> It's due to the required semantics of the component.  `<RecoilSync>` should be an ancestor component that encompasses components which use the atoms that are synced with that store.  If it was just a hook then the hook could potentially be called after the nested component using the synced atom was called.

Makes sense. Thanks!
